### PR TITLE
[CC-52142] Add tests for data sources and resource import

### PR DESCRIPTION
**Pull request by erademacher**
Made sure all resources and data sources have tests. Added import steps to resources that didn't already have one.

**Commit checklist**
- [n/a] Changelog
- [n/a] Doc gen (`make generate`)
- [x] Integration test(s)
- [x] Acceptance test(s)
- [n/a] Example(s)

### CODE REVIEW

File: internal/provider/cluster_cert_data_source_test.go [hunk 1]
```diff
@@ -0,0 +1,121 @@
+/*
+ Copyright 2022 The Cockroach Authors
```
Comment (sjbarag) [inline internal/provider/cluster_cert_data_source_test.go:2]: nit: it is unfortunately 2023 🕐 

where do the years go 😢

File: internal/provider/cluster_cert_data_source_test.go [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+	mock_client "github.com/cockroachdb/terraform-provider-cockroach/mock"
+	"github.com/golang/mock/gomock"
+	"github.com/google/uuid"
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/stretchr/testify/require"
+)
+
+// TestAccClusterCertResource attempts to download a cluster's cert.
+// It will be skipped if TF_ACC isn't set.
```
Comment (sjbarag) [inline internal/provider/cluster_cert_data_source_test.go:35]: Thanks for the reminder! It's super helpful :)

File: internal/provider/connection_string_data_source_test.go [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+			Serverless: &client.ServerlessClusterConfig{},
+		},
+		Regions: []client.Region{
+			{
+				Name: "us-central1",
+			},
+		},
+	}
+
+	db, sqlUser, os := "testdb", "test", "MAC"
```
Comment (sjbarag) [inline internal/provider/connection_string_data_source_test.go:71]: I dunno why I've never thought to do this, but it makes so much sense 🤯

File: internal/provider/connection_string_data_source_test.go [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+	s.EXPECT().GetConnectionString(gomock.Any(), clusterId, connectStringOptions).
+		Return(connectionStringResponse, httpOkResponse, nil).Times(4)
+	s.EXPECT().DeleteCluster(gomock.Any(), clusterId)
+
+	testConnectionStringDataSource(t, clusterName, true)
+}
+
+func testConnectionStringDataSource(t *testing.T, clusterName string, useMock bool) {
+	connectStrDataSourceName := "data.cockroach_connection_string.test"
+	sqlPassword := "hunter2"
```
Comment (sjbarag) [inline internal/provider/connection_string_data_source_test.go:98]: erm, all I see is `sqlPassword := "*******"`. Is github smart enough to know that `hunter2` is your password and show me `*******` instead?

**Review by sjbarag — APPROVED** [review]
Looks great @erademacher! Thanks again for the walkthrough yesterday 😄 

(It is, alas, 2023 so we should probably get the license headers in this PR updated. Approved on the assumption you'll do that before you merge.)

File: internal/provider/connection_string_data_source_test.go (hunk 4 above)
Comment (erademacher) [reply internal/provider/connection_string_data_source_test.go:98]: Yeah, I was a little worried after I realized that I used my actual 1Password master password here, but then I remembered that GitHub's got my back

**Comment by erademacher** [comment]
TFTR! I'll make a followup to update the rest of the boilerplate.

**Comment by jenngeorge** [comment]
also LGTM !

**Pull request merged**